appveyor.yml

```diff
@@ -1,11 +1,65 @@
 image: Visual Studio 2015
 
+environment:
+  NUGET_API_KEY:
+    secure: Rh/UHYn7HGj8a7LFWo9C1rSA0sHSQNV3wIqj9bvwRR8yLmCXu9JmbNkmkctk2cBn
+  MYGET_API_KEY:
+    secure: hA4Ut1N2lrrdEtAN24Bty/FNiU0d/Ur/dLYSqpr8jKHOvoO7MU4jD+KwzUvATh+E
+
 pull_requests:
   do_not_increment_build_number: true
 
 build_script:
 - ps: |
-    & .\build.cmd CompleteBuild
+    # A trick to test if variable is defined. Number could never be null.
+    $isPullRequest = -not ($Env:APPVEYOR_PULL_REQUEST_NUMBER -eq $null)
+    $isTagBuild = $Env:APPVEYOR_REPO_TAG -eq "true"
+    $branchName = $Env:APPVEYOR_REPO_BRANCH
+
+    Write-Host "[BUILD CONTEXT INFO] Is PR: $isPullRequest, is tag: $isTagBuild, branch name: '$branchName'" -ForegroundColor Green
+
+    if($isTagBuild)
+    {
+      $tagName = $Env:APPVEYOR_REPO_TAG_NAME
+      $isSemVerTag = $tagName -match '^v.*'
+      # We consider version as alpha if it contains the "-alpha" pre-release tag.
+      $isAlphaVerTag = $tagName -match '^v[\d\.]+-alpha[\d\.]*$'
```

Comment:
Do we really have to create a tag in order to publish to the private feed? I'm afraid that is going to leave us with _a lot_ of tags over time. 😟 Can't we simply publish all builds from the `vNext` branch to the private feed, instead? The branch name could be included as _pre-release_ tag in the version number to differentiate the packages.